test/topology_custom/test_mv_topology_change.py

```diff
@@ -77,3 +80,63 @@ async def do_writes(start_it, repeat) -> int:
     stop_event.set()
     await asyncio.gather(*tasks)
 
+# Reproduces #19152
+# Verify a pending replica is not doing unnecessary work of building and sending view updates.
+# 1) we have a table with a materialized view
+# 2) start migrating a tablet from node 1 to node 2
+# 3) while node 2 is a pending replica, write to the table
+# 4) complete migration
+# 5) verify node 2 did not build view updates
+@pytest.mark.asyncio
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_mv_update_on_pending_replica(manager: ManagerClient):
+    cfg = {'enable_tablets': True}
+    servers = [await manager.server_add(config=cfg)]
+
+    await manager.api.disable_tablet_balancing(servers[0].ip_addr)
+
+    cql = manager.get_cql()
+    await cql.run_async("CREATE KEYSPACE test WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1} AND tablets = {'initial': 1};")
+    await cql.run_async("CREATE TABLE test.test (pk int PRIMARY KEY, c int);")
+    await cql.run_async("CREATE MATERIALIZED VIEW test.mv1 AS SELECT * FROM test.test WHERE pk IS NOT NULL AND c IS NOT NULL PRIMARY KEY (c, pk);")
+
+    servers.append(await manager.server_add(config=cfg))
+
+    key = 7
+    tablet_token = 0
+    await cql.run_async(f"INSERT INTO test.test (pk, c) VALUES ({key}, 0)")
+
+    replica = await get_tablet_replica(manager, servers[0], 'test', 'test', tablet_token)
+
+    s0_host_id = await manager.get_host_id(servers[0].server_id)
+    s1_host_id = await manager.get_host_id(servers[1].server_id)
+    dst_shard = 0
+
+    await manager.api.enable_injection(servers[1].ip_addr, "stream_mutation_fragments", one_shot=True)
+    s1_log = await manager.server_open_log(servers[1].server_id)
+    s1_mark = await s1_log.mark()
+
+    migration_task = asyncio.create_task(
+        manager.api.move_tablet(servers[0].ip_addr, "test", "test", replica[0], replica[1], s1_host_id, dst_shard, tablet_token))
+
+    await s1_log.wait_for('stream_mutation_fragments: waiting', from_mark=s1_mark)
+    s1_mark = await s1_log.mark()
```

Comment:
An idea: perhaps instead of checking the log, we could directly check somehow (system table?) that the second node is now a pending replica? But I don't insist. I don't know how difficult it is to do.

Reply:
I changed the test to check the streaming state in the system table instead of logs